cupyx/lapack.py

```diff
@@ -50,15 +50,15 @@ def gesv(a, b):
 
     n = b.shape[0]
     nrhs = b.shape[1] if b.ndim == 2 else 1
-    if a._c_contiguous:
+    if a._f_contiguous:
+        trans = _cublas.CUBLAS_OP_N
+    elif a._c_contiguous:
         trans = _cublas.CUBLAS_OP_T
     else:
-        trans = _cublas.CUBLAS_OP_N
-        if not a._f_contiguous:
-            a = a.copy(order='F')
-    ret_b = b
-    if nrhs > 1 and not ret_b._f_contiguous:
-        b = ret_b.copy(order='F')
+        raise ValueError('a must be F-contiguous or C-contiguous.')
+    if not (b._f_contiguous or (b._c_contiguous and nrhs == 1)):
+        raise ValueError('b must be F-contiguous '
+                         '(or C-contiguous if nrhs == 1).')
```

Comment:
The message looks good.

For the `nrhs == 1` cases, I suppose `b._f_contiguous` iff `b._c_contiguous`.

Reply:
Thanks, I did not know that.
Should I change (simplify) the `if` condition?

Reply:
Shorter error message looks better here. Applying `asfortranarray` to a C- and F-contiguous array is redundant, but has no GPU cost.